### PR TITLE
Continuation of package splitting

### PR DESCRIPTION
**Pull request by hobu**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@hobu @chambbj , this is ready for a review. Current plan is to

1. Create a libpdal-feedstock with the recipe in the PR by submitting to staged-recipes
2. Build python-pdal in python-pdal-feedstock using libpdal
3. Make this feedstock create a `pdal==2.7.1` meta-package that installs `libpdal-all` and `python-pdal`.

Let me know what you think

File: recipe/meta.yaml [hunk 1]
```diff
@@ -207,6 +207,7 @@ outputs:
       host:
         - {{ pin_subpackage('libpdal', exact=True) }}
         - xerces-c
+        - nlohmann_json
```
Comment (hobu) [inline recipe/meta.yaml:211]: There should be no external linkage of nlohmann in any of PDAL's plugins. PDAL vendors its own copy and that should be used.

**Comment by hobu** [comment]
I will upstream all of these patches to PDAL's `2.7-maintenance` branch and they will be available when 2.7.2 is released.

**Comment by hobu** [comment]
What will happen to pdal-feedstock after implementing this plan? Will it be removed?

**Comment by isuruf** [comment]
> What will happen to pdal-feedstock after implementing this plan? Will it be removed?

No, it will produce a `pdal` meta-package that installs `libpdal-all` and `python-pdal`

**Comment by isuruf** [comment]
If `libgdal` is going to be further split up to `libgdal-core` + plugins, we might have to change the naming here.

**Comment by hobu** [comment]
> If libgdal is going to be further split up to libgdal-core + plugins, we might have to change the naming here.

Indeed, we want to follow whatever GDAL does.

I've looked this over and I don't have any complaints. I have merged up your patches to the PDAL 2.7 branch and they will be available when I release PDAL 2.7.2 middle of next week.

I'm 👍 to merge whenever you are ready.

**Pull request closed without merging**